tests/unit/test_patch.py

```diff
@@ -0,0 +1,204 @@
+import json
+
+from dash import Patch
+from dash._utils import to_json
+
+
+def patch_to_dict(p):
+    return json.loads(to_json(p))
+
+
+def test_pat001_patch_assign_item():
+    p = Patch()
+    p["item"] = "item"
+
+    data = patch_to_dict(p)
+
+    assert data["operations"][0] == {
+        "operation": "Assign",
+        "location": ["item"],
+        "params": {"value": "item"},
+    }
+
+
+def test_pat002_patch_assign_attr():
+    p = Patch()
+    p.item = "item"
+
+    data = patch_to_dict(p)
+
+    assert data["operations"][0] == {
+        "operation": "Assign",
+        "location": ["item"],
+        "params": {"value": "item"},
+    }
+
+
+def test_pat003_patch_multi_operations():
+    p = Patch()
+    p.one = 1
+    p.two = 2
+
+    data = patch_to_dict(p)
+
+    assert len(data["operations"]) == 2
+    assert data["operations"][0]["location"] == ["one"]
+    assert data["operations"][1]["location"] == ["two"]
+
+
+def test_pat004_patch_nested_assign():
+    p = Patch()
+
+    p["nest_item"]["nested"]["deep"] = "deep"
+    p.nest_attr.nested.deep = "deep"
+
+    data = patch_to_dict(p)
+
+    assert data["operations"][0]["location"] == ["nest_item", "nested", "deep"]
+    assert data["operations"][1]["location"] == ["nest_attr", "nested", "deep"]
+
+
+def test_pat005_patch_delete_item():
+    p = Patch()
+
+    del p["delete_me"]
```

Comment:
Does it also work to remove an item from a list? ie `del p[5]` or `del p[-1]`?

Reply:
Should work it's dissocPath, but let's add a test.